File: app/src/main/java/com/example/screendesign2/adapters/SliderPageAdapter.java
package com.example.screendesign2.adapters;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.media.Image;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.core.content.res.ResourcesCompat;
import androidx.recyclerview.widget.RecyclerView;
import androidx.viewpager.widget.PagerAdapter;

import com.example.screendesign2.R;
import com.google.android.material.slider.Slider;

import java.util.List;
import java.util.Objects;

public class SliderPageAdapter extends RecyclerView.Adapter<SliderPageAdapter.ViewHolder> {
    int[] images;
    Context context;


    public SliderPageAdapter(Context context, int[] images){
        this.context = context;
        this.images = images;
    }


    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(context);

        // Inflate the custom layout
        View imageView = inflater.inflate(R.layout.slider_image_view, parent, false);

        // Return a new holder instance
        ViewHolder viewHolder = new ViewHolder(imageView);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        int image = images[position];

        // Set item views based on your views and data model
        ImageView imageView = holder.mainImage;
        Drawable drawable = context.getResources().getDrawable(image);
        imageView.setImageDrawable(drawable);

    }

    @Override
    public int getItemCount() {
        return images.length;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

    public ImageView mainImage;


    public ViewHolder(@NonNull View itemView) {
        super(itemView);
        mainImage = (ImageView) itemView.findViewById(R.id.imageViewMain);

    }
}
}

File: app/src/main/java/com/example/screendesign2/adapters/SliderPageAdapter.kt
package com.example.screendesign2.adapters

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import com.example.screendesign2.R
import android.widget.TextView
import com.example.screendesign2.data.Slider

class SliderPageAdapter(var context: Context, var sliders: List<Slider>) :
    RecyclerView.Adapter<SliderPageAdapter.ViewHolder>() {
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val inflater = LayoutInflater.from(context)

        // Inflate the custom layout
        val imageView =
            inflater.inflate(R.layout.slider_image_view, parent, false)

        // Return a new holder instance
        return ViewHolder(imageView)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val image = sliders[position].image
        val title = sliders[position].title
        val subTitle = sliders[position].subTitle

        // Set item views based on your views and data model
        val imageView = holder.mainImage
        val drawable = context.resources.getDrawable(image)
        imageView.setImageDrawable(drawable)
        val mainTextTitle = holder.mainTextTitle
        mainTextTitle.text = title
        val mainTextSubTitle = holder.mainTextSubTitle
        mainTextSubTitle.text = subTitle
    }

    override fun getItemCount(): Int {
        return sliders.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var mainImage: ImageView
        var mainTextTitle: TextView
        var mainTextSubTitle: TextView

        init {
            mainImage = itemView.findViewById<View>(R.id.imageViewMain) as ImageView
            mainTextTitle = itemView.findViewById<View>(R.id.textViewMain) as TextView
            mainTextSubTitle = itemView.findViewById<View>(R.id.textViewSubTitle) as TextView
        }
    }
}